server.R

```diff
@@ -103,4 +103,143 @@ function(input, output, session) {
+  value <- reactiveValues(etaGSD = 0) # create reactive variables for dilution effect eta
+  value <- reactiveValues(psiGSD = 1) # create reactive variables for variance inflation psi
+  
+  output$dilutionSettings <- renderUI({ # show input for eta if box is unticked
+    if(input$dilutionCheck == FALSE) {
+      sliderInput("etaGSD", label=HTML("Dilution effect &eta;"), min = 0, max = 1, value = 0)
+    }
+  })
+  observeEvent(input$etaGSD, { # update eta according to slider
+    if(input$dilutionCheck == FALSE) value$etaGSD <- input$etaGSD
+  })
+  observeEvent(input$dilutionCheck, { # update eta if box is ticked again
+    if(input$dilutionCheck == TRUE) value$etaGSD <- 0
+  })
+  
+  output$psiSettings <- renderUI({ # show input for psi if box is unticked
+    if(input$psiCheck == FALSE) {
+      numericInput("psiGSD", label=HTML("Variance inflation &psi;"), min = 0.1, max = 5, value = 1, step = 0.1)
+    }
+  })
+  observeEvent(input$psiGSD, { # update psi according to input
+    if(input$psiCheck == FALSE) value$psiGSD <- input$psiGSD
+  })
+  observeEvent(input$psiCheck, { # update psi if box is ticked again
+    if(input$psiCheck == TRUE) value$psiGSD <- 1
+  })
+  
+ 
+  powerGSD <- reactive({ # Create table for the resulting values of the power (independent of input value for tau)
+    # values for tau
+    TAU = c(seq(0.05, 0.95, 0.05), 0.99)*100
+    # matrix for results
+    data.gsd = matrix(NA, ncol=4, nrow=length(TAU))
+    for(tauGSD in TAU) {
+      # get critical valus for GSD from package rpact
+      crit = getDesignGroupSequential(
+        kMax = 2,
+        typeOfDesign = input$designSelect, 
+        informationRates = c(tauGSD/100, 1),
+        alpha = input$alphaGSD, 
+        sided = 1)$criticalValues
+      # calculate means for first stage and final stage test statistics
+      mu.t0 = (qnorm(1-input$alphaGSD)+qnorm(input$powerGSD/100)) * sqrt(tauGSD/100)
+      mu.t =  (qnorm(1-input$alphaGSD)+qnorm(input$powerGSD/100)) * (tauGSD/100+(1-tauGSD/100)*(1-value$etaGSD))/sqrt(tauGSD/100 + (1-tauGSD/100)*value$psiGSD)
+      # calculate variance matrix for the test statistics
+      SIG = matrix(c(1, 
+                     sqrt(tauGSD/100*1/(tauGSD/100 + (1-tauGSD/100)*value$psiGSD)), 
+                     sqrt(tauGSD/100*1/(tauGSD/100 + (1-tauGSD/100)*value$psiGSD)), 
+                     1), ncol=2)
+      # calculate power for fixed design and GSD (first stage and final stage)
+      power.fix = 1-pnorm(qnorm(1-input$alphaGSD), mean = mu.t0 , sd=1)
+      power.gsd.1 = 1-pnorm(crit[1], mean = mu.t0, sd=1)
+      power.gsd.2 = 1-pmvnorm(lower = c(-Inf, -Inf), upper = c(crit[1], crit[2]), 
+                              mean = c(mu.t0 , mu.t), 
+                              sigma = SIG)[1]
+      data.gsd[which(tauGSD==TAU), 1:4] = c(tauGSD, power.fix*100, power.gsd.1*100, power.gsd.2*100)
+    }
+    return(data.gsd)
+  })
+  
+  powerGSDinput <- reactive({ # Add row for input value of tau

```

Comment:
When I chose 60% using the slider, I got two rows for 60%; but this doesn't happen to, say 50%.